### PR TITLE
Lazy conversion from backend kil to generic kil for faster step and resume. 

### PR DESCRIPTION
**Pull request by msaxena2**
Faster debugger step and resume due to lazy conversion from backend kil to generic kil.

### CODE REVIEW

File: java-backend/src/main/java/org/kframework/backend/java/util/JavaKRunState.java [hunk 1]
```diff
@@ -40,7 +40,8 @@ public Term getRawResult() {
     @Override
     public boolean equals(Object o) {
         if (!(o instanceof JavaKRunState)) {
-            return false;
+            KRunState genericObj = (KRunState) o;
```
Comment (dwightguth) [inline java-backend/src/main/java/org/kframework/backend/java/util/JavaKRunState.java]: this is not a valid implementation of equals. For example, it is not symmetric. I refer you to the javadoc for Object.equals: http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html#equals%28java.lang.Object%29

Another point to note is that this function now violates the contract of hashCode. In particular, hashCode guarantees that if two objects compare equal with equals(), they must have the same hash code.

I think probably the simplest solution to this problem is to get rid of the GenericKRunState class and replace instances of its constructor with `executor.step(term, 0, false).getFinalState()`

**Comment by msaxena2** [comment]
Jenkins: test this please!

**Comment by msaxena2** [comment]
Jenkins: test this please

**Comment by msaxena2** [comment]
Jenkins: test this please

**Pull request merged**